fluent-syntax/src/ast.js

```diff
@@ -210,13 +210,6 @@ export class Section extends Entry {
   }
 }
 
-export class Function extends Identifier {
```

Comment:
Removing this doesn't look right. `Function` is a valid ASDL production: https://github.com/projectfluent/fluent/blob/a9fb740ecbf9326a484b812800f62a338a43f7da/spec/fluent.asdl#L72

Instead, I think when we create a `CallExpression`, we should ensure that the `callee` field is a `Function`, not just an `Identifier`: https://github.com/projectfluent/fluent.js/blob/3e8066cfa81b31b4d46aec99a74e477792c61f5b/fluent-syntax/src/parser.js#L569